urwid/display/_raw_display_base.py

```diff
@@ -640,14 +642,17 @@ def using_standout_or_underline(a: AttrSpec | str) -> bool:
             first = True
             lasta = lastcs = None
             for a, cs, run in row:
-                if not isinstance(run, bytes):  # canvases should render with bytes
+                if not isinstance(run, bytes):  # canvases render with bytes
                     raise TypeError(run)
+
                 if cs != "U":
                     run = run.translate(UNPRINTABLE_TRANS_TABLE)  # noqa: PLW2901
+
                 if first or lasta != a:
                     o.append(attr_to_escape(a))
                     lasta = a
-                if first or lastcs != cs:
+
+                if not (IS_WINDOWS or IS_WSL) and (first or lastcs != cs):
```

Comment:
technically it's not needed at all if console is Unicode but need deeper testing.